astropy/table/pprint.py

```diff
@@ -266,51 +266,27 @@ def _pformat_col(self, col, max_lines=None, show_name=True, show_unit=None,
         else:
             col_width = max(len(x) for x in col_strs) if col_strs else 1
 
-            # Center line content and generate dashed headerline
+            # Center line header content and generate dashed headerline
             for i in outs['i_centers']:
                 col_strs[i] = col_strs[i].center(col_width)
             if outs['i_dashes'] is not None:
                 col_strs[outs['i_dashes']] = '-' * col_width
 
-            # Format columns according to alignment
-            justify_methods = {'<': 'ljust', '^': 'center', '>': 'rjust'}
+            # Format columns according to alignment.  `align` arg has precedent, otherwise
+            # use `col.format` if it is a legal alignment string.  If neither applies
+            # then right justify.
+            justify_methods = {'<': 'ljust', '^': 'center', '>': 'rjust', '0=': 'zfill'}
+            align = align or (col.info.format
+                              if col.info.format in justify_methods
+                              else '>')
+
+            # This can only occur if `align` was explicitly provided.
+            if align not in justify_methods:
```

Comment:
@taldcroft - this leaves much less freedom to use the alignment than the admittedly long if-else clause that you removed, in that previously formats like `>8.3f` would work, as would `.<20.3f` (which would put dots to fill up any space). E.g., with the implementation as is one can get ApJ-table style effects as follows:
```
In [5]: t = Table([['src 1', 'src 2 with a long name']], names=['a'])

In [6]: t['a'].format = '.<'

In [7]: t
Out[7]: 
<Table length=2>
          a           
        str22         
----------------------
src 1.................
src 2 with a long name
```

Can we keep the long statement but just move it out of the loop? (There is no reason for it to be inside.) Though for the case that `align` is defined, the big construction is needless; don't know how that passed by me...
```
    else:
        justify_method = justify_methods.get(align,